### ISSUE TITLE
How to know <PYTORCH_PARAM_NAME_STR>?

### ISSUE THREAD

**Issue by bupt-renpei** [NervanaSystems/distiller#471]
Dear all, when we define the scheduler, we need to specify the DNN parameter name, but how can we know their name (e.g., module.layer3.1.conv1.weight) for a given DNN model?

regularizers:
  my_L1_reg:
    class: L1Regularizer
    reg_regims:
      'module.layer3.1.conv1.weight': 0.000002
      'module.layer3.1.conv2.weight': 0.000002
      'module.layer3.1.conv3.weight': 0.000002
      'module.layer3.2.conv1.weight': 0.000002
    threshold_criteria: Mean_Abs

**Comment by nzmora** [comment]
Hi @bupt-renpei,

The names of the parameters follow the PyTorch convention.  You can print the parameters of a PyTorch model like so:
```
for name, param in model.named_parameters():
    print(name)
```

or like so:
```
for name, param in model.state_dict().items():
    print(name)
```

Note that PyTorch adds the `module.` prefix to the names of modules that are wrapped by a `nn.DataParallel` module.  Distiller wraps models with `nn.DataParallel` by default using this code:https://github.com/NervanaSystems/distiller/blob/db597222c442ca22bd7bfe79774640a7eef5bd9c/distiller/models/__init__.py#L144-L148 

You can also use Distiller's `summary` utilities https://github.com/NervanaSystems/distiller/blob/db597222c442ca22bd7bfe79774640a7eef5bd9c/distiller/model_summaries.py#L82

as exemplified here: https://github.com/NervanaSystems/distiller/wiki/Distiller-Summary-Reports#sparsity

Cheers,
Neta
